pycmds/_record.py

```diff
@@ -266,55 +215,21 @@ def create_data(self, inputs):
         # create folder
         data_path.write(os.path.join(scan_folder, self.filename + ".data"))
         # generate file
-        dictionary = headers.read(kind="data")
+        dictionary = headers.read()
         tidy_headers.write(data_path.read(), dictionary)
-        # shots ---------------------------------------------------------------
-        # TODO: this is hack
-        if aqn.has_section("NI 6251"):
-            if False and aqn.read("NI 6251", "save shots"):
-                p = os.path.join(os.path.dirname(data_path.read()), "NI 6251 shots.hdf5")
-                f = h5py.File(p)
-                dictionary = headers.read(kind="shots")
-                for key, value in dictionary.items():
-                    # remove None
-                    if type(value) is list:
-                        for i, val in enumerate(value):
-                            if val is None:
-                                value[i] = "None"
-                            if isinstance(val, basestring):
-                                value[i] = str(val)  # cannot handle unicode in lists...
-                    # write to hdf5
-                    f.attrs[key] = value
-                col_count = len(dictionary["name"])
-                f.create_dataset(
-                    "array", (col_count, 0), maxshape=(col_count, None), compression="gzip",
-                )
-                f["array"].set_fill_value = np.nan
-                f.close()
 
     def write_data(self, inputs):
-        data_arr, shots_arr = inputs
+        data_arr = inputs[0]
         # pixels --------------------------------------------------------------
         data_file = open(data_path.read(), "ab")
-        if len(data_arr.shape) == 2:  # case of multidimensional devices
+        if len(data_arr.shape) == 2:  # case of multidimensional sensors
```

Comment:
not new... but `data_arr.ndim`, and should we be handling dimensions > 2?

Reply:
later